src/configr_assign.hpp

```diff
@@ -0,0 +1,46 @@
+/*
+   Copyright (C) 2014 - 2016 by David White <dave@whitevine.net>
+   Part of the Battle for Wesnoth Project http://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+#pragma once
+
+#include <string>
+#include "config.hpp"
+#include <boost/variant.hpp>
+
+struct configr_of
```

Comment:
r for reference since it, unlike config_of just references a config. feel free to suggest a better name.